packages/arb-token-bridge-ui/src/token-bridge-sdk/EthDepositStarter.ts

```diff
@@ -85,35 +86,78 @@ export class EthDepositStarter extends BridgeTransferStarter {
     // no-op
   }
 
-  public async transferEstimateGas({ amount, signer }: TransferEstimateGas) {
+  public async transferEstimateGas({
+    amount,
+    signer,
+    destinationAddress
+  }: TransferEstimateGas) {
     const address = await getAddressFromSigner(signer)
 
     return depositEthEstimateGas({
       amount,
       address,
       parentChainProvider: this.sourceChainProvider,
-      childChainProvider: this.destinationChainProvider
+      childChainProvider: this.destinationChainProvider,
+      destinationAddress
     })
   }
 
-  public async transfer({ amount, signer }: TransferProps) {
+  public async transfer({ amount, signer, destinationAddress }: TransferProps) {
     const address = await getAddressFromSigner(signer)
     const ethBridger = await this.getBridger()
 
-    const depositRequest = await ethBridger.getDepositRequest({
-      amount,
-      from: address
-    })
+    const isDifferentDestinationAddress =
+      destinationAddress &&
+      destinationAddress.toLowerCase() !== address.toLowerCase()
+
+    // TODO: remove this when eth-custom-dest feature is live
+    // this is a safety check, this shouldn't happen
+    if (
+      isDifferentDestinationAddress &&
+      !isExperimentalFeatureEnabled('eth-custom-dest')
+    ) {
+      throw 'Native currency transfers to a custom destination address are not supported yet.'
+    }
+
+    const depositRequest = isDifferentDestinationAddress
+      ? await ethBridger.getDepositToRequest({
+          amount,
+          from: address,
+          parentProvider: this.sourceChainProvider,
+          childProvider: this.destinationChainProvider,
+          destinationAddress
+        })
+      : await ethBridger.getDepositRequest({
+          amount,
+          from: address
+        })
 
     const gasLimit = await this.sourceChainProvider.estimateGas(
       depositRequest.txRequest
     )
 
-    const sourceChainTransaction = await ethBridger.deposit({
-      amount,
-      parentSigner: signer,
-      overrides: { gasLimit: percentIncrease(gasLimit, BigNumber.from(5)) }
-    })
+    const parentChainOverrides = {
+      gasLimit: percentIncrease(gasLimit, BigNumber.from(5))
+    }
+
+    const sourceChainTransaction = destinationAddress
```

Comment:
We haven't added a check for `isDifferentDestinationAddress` here. This will pass same-destination transfer into custom destination transfer flow for `destinationAddress === walletAddress`.

Perhaps we should separate the flows for `isDifferentDestinationAddress` from L122 to L160 so that code is better partitioned?

Reply:
I've updated it.

I think it's fine to keep it together? Otherwise we will have to repeat ourselves with some of the code.